test/integration/cli/test/index.test.js

```diff
@@ -579,6 +579,27 @@ describe('CLI Usage', () => {
       expect(stripAnsi(stdout).trim()).toBeFalsy()
     })
 
+    test('Allow retry if port 3000 is already in use', async () => {
+      const port = 3000
```

Comment:
Can we have a test for the non-default port as well just for sanity?

Reply:
We already have that here → https://github.com/vercel/next.js/blob/c5b54508a11a09dcd37e49e39712118d1f519093/test/integration/cli/test/index.test.js#L555, https://github.com/vercel/next.js/blob/c5b54508a11a09dcd37e49e39712118d1f519093/test/integration/cli/test/index.test.js#L571 (this throws the error when random specified is already in use)

Reply:
Oh this seems wrong, if a manual port is provided it should always error if the port is already taken, it should only retry finding a different port if the port argument wasn't provided

Reply:
Isn't that what's it's doing? https://github.com/vercel/next.js/blob/c5b54508a11a09dcd37e49e39712118d1f519093/test/integration/cli/test/index.test.js#L578

Reply:
Can we ensure if `-p` is provided regardless of the port even if default it doesn't attempt retrying to avoid a breaking change?

I know some flows use this to avoid accidentally starting on a new port when a previous instance is still running which can make debugging confusing.